From 8f59350c1f860edb6344995edb4b822cf24e140c Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Mon, 2 Mar 2015 17:52:27 +0800
Subject: [PATCH] MLK-10257-01 arm: dts: Add operating-points property for cpu
 node

Add operating-points, clks and regulators property used in DVFS

Signed-off-by: Bai Ping <b51503@freescale.com>
---
 arch/arm/boot/dts/imx7d-12x12-arm2.dts | 4 ++++
 arch/arm/boot/dts/imx7d-19x19-arm2.dts | 4 ++++
 arch/arm/boot/dts/imx7d-sdb.dts        | 4 ++++
 arch/arm/boot/dts/imx7d.dtsi           | 9 +++++++++
 4 files changed, 21 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-12x12-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
index 7f6dcca..cf2ba93 100644
--- a/arch/arm/boot/dts/imx7d-12x12-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
@@ -129,6 +129,10 @@
 	};
 };
 
+&cpu0 {
+	arm-supply = <&sw1a_reg>;
+};
+
 &ecspi3 {
 	fsl,spi-num-chipselects = <1>;
 	cs-gpios = <&gpio6 22 0>;
diff --git a/arch/arm/boot/dts/imx7d-19x19-arm2.dts b/arch/arm/boot/dts/imx7d-19x19-arm2.dts
index e424d5b..6d6de4b 100644
--- a/arch/arm/boot/dts/imx7d-19x19-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-19x19-arm2.dts
@@ -125,6 +125,10 @@
 	};
 };
 
+&cpu0 {
+	arm-supply = <&sw1a_reg>;
+};
+
 &fec1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_enet1>;
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index f19c569..80131af 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -112,6 +112,10 @@
 	};
 };
 
+&cpu0 {
+	arm-supply = <&sw1a_reg>;
+};
+
 &epdc {
         pinctrl-names = "default";
         pinctrl-0 = <&pinctrl_epdc0>;
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index 65ee777..66f1cee 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -49,6 +49,15 @@
 			compatible = "arm,cortex-a7";
 			device_type = "cpu";
 			reg = <0>;
+			operating-points = <
+				/* KHz	uV */
+				996000	1075000
+				792000	975000
+			>;
+			clock-latency = <61036>; /* two CLK32 periods */
+			clocks = <&clks IMX7D_ARM_A7_ROOT_CLK>, <&clks IMX7D_ARM_A7_ROOT_SRC>,
+				 <&clks IMX7D_PLL_ARM_MAIN_CLK>, <&clks IMX7D_PLL_SYS_MAIN_CLK>;
+			clock-names = "arm", "arm_root_src", "pll_arm", "pll_sys_main";
 		};
 
 		cpu1: cpu@1 {
-- 
1.8.0

